src/main/webapp/app/core/auth/account.service.ts

```diff
@@ -21,15 +22,15 @@ export interface IAccountService {
     isAtLeastTutorInCourse: (course: Course) => boolean;
     isAtLeastInstructorInCourse: (course: Course) => boolean;
     isAuthenticated: () => boolean;
-    getAuthenticationState: () => Observable<User>;
+    getAuthenticationState: () => Observable<User | null>;
     getImageUrl: () => string | null;
 }
 
 @Injectable({ providedIn: 'root' })
 export class AccountService implements IAccountService {
-    private userIdentity: User | null;
+    private userIdentityValue: User | null = null;
```

Comment:
why do we need `null = null`?

Reply:
This gives the userIdentity the inital value `null`, if I would not do this, the initial value would be `undefined`.
This would then represent a third state "has not logged in yet" besides "logged in" (`typeof User`) and "logged out" (`null`) that we have no use case for atm.

As the `authenticationState` subject should only emit changes (e.g. `null => User => null` vs. `null => null => User => null`), this would complicate the logic a bit as we would have to differentiate changes as `undefined => null => User => null` where the first two emits will not make a difference for the subscribers.

Reply:
ok, I understand. It looks just weird how this is written. I'm not used to type definitions such as `User | null`. Could we not just write 
`private userIdentityValue: User? = null;`

or would this imply undefined as well?

Reply:
I merge this now, we can talk about this on Tuesday